.github/workflows/amazon_bedrock.yml

```diff
@@ -62,7 +62,23 @@ jobs:
         if: matrix.python-version == '3.9' && runner.os == 'Linux'
         run: hatch run docs
 
+      - name: Set IS_FORK environment variable (Unix-like)
+        if: runner.os != 'Windows'
+        run: |
+          if [ "${{ github.event.pull_request.head.repo.full_name }}" != "${{ github.repository }}" ]; then
+            echo "IS_FORK=true" >> $GITHUB_ENV
+          fi
+
+      - name: Set IS_FORK environment variable (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          if ($Env:GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FULL_NAME -ne $Env:GITHUB_REPOSITORY) {
+            echo "IS_FORK=true" >> $Env:GITHUB_ENV
+          }
+
+
       - name: AWS authentication
+        if: env.IS_FORK != 'true'      
```

Comment:
Maybe it's more complex than this, but why don't we split the tests in unit and integration, then we put `if: github.event.pull_request.head.repo.full_name == github.repository` here and in the integration tests step?

Reason is I'm not a fan of making the testing code CI dependent.

Reply:
Makes sense. I will try!